### PR TITLE
[Merged by Bors] - feat(Data/Int/WithZero): add WithZeroMultIntToNNReal

### PR DESCRIPTION
**Pull request by mariainesdff**
We define  the morphism `withZeroMultIntToNNReal`  from `ℤₘ₀ → ℝ≥0` sending `0 ↦ 0` and  `x ↦ e^(Multiplicative.toAdd (WithZero.unzero hx)` when `x ≠ 0`, for a nonzero `e : ℝ≥0` and prove some of its properties.

Co-authored-by: Filippo A. E. Nuccio <filippo.nuccio@univ-st-etienne.fr>

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Int/WithZero.lean [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+WithZero, multiplicative, nnreal
+-/
+
+noncomputable section
+
+open scoped NNReal
+
+open Multiplicative WithZero
+
+local notation "ℤₘ₀" => WithZero (Multiplicative ℤ)
```
Comment (jcommelin) [inline Mathlib/Data/Int/WithZero.lean:35]: The names of these theorems become pretty long. Would it make sense to make this notation into an `abbrev` and then use the `WithZeroMulInt` namespace. Then the morphism below would become `WithZeroMulInt.toNNReal`.

Comment (mariainesdff) [reply Mathlib/Data/Int/WithZero.lean]: Done!

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

File: Mathlib/Data/Int/WithZero.lean [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+noncomputable section
+
+open scoped NNReal
+
+open Multiplicative WithZero
+
+/-- The type `WithZero (Multiplicative ℤ)`. -/
+abbrev WithZeroMulInt := WithZero (Multiplicative ℤ)
+
+local notation "ℤₘ₀" => WithZeroMulInt
```
Comment (YaelDillies) [inline Mathlib/Data/Int/WithZero.lean:38]: This notation is already defined by this point (you might have to merge master)
```suggestion
```

**Comment by sgouezel** [comment]
Can you merge master, fix the build, and then send it back to bors?
bors d+

**Pull request closed without merging**